src/os.c

```diff
@@ -237,13 +242,37 @@ void _mi_os_init() {
 
 #else  // generic unix
 
+static void os_detect_transparent_huge_pages(void) {
+#if defined(MADV_HUGEPAGE)
+  int fd = -1;
+  size_t i;
+  static const char *paths[] = {
+      "/sys/kernel/mm/transparent_hugepage/enabled",
+      "/sys/kernel/mm/redhat_transparent_hugepage/enabled",
+  };
+
+  for (i = 0; i < sizeof(paths) / sizeof(paths[0]); i ++) {
+      fd = open(paths[i], O_RDONLY);
+      if (fd != -1) break;
+  }
+
+  if (fd < 0) return;
+  char buf[128];
+  ssize_t nread = read(fd, &buf, sizeof(buf));
+  close(fd);
+  if (nread >= 1) {
+      if (strstr(buf, "[madvise]")) os_transparent_huge_pages = true;
```

Comment:
`buf` is not null-terminated, so `strstr` is undefined behavior. `memmem(buf, nread, "[madvise]", strlen("[madvise]"))` instead?